src/main/java/com/example/bank/registration/ClientController.java

```diff
@@ -0,0 +1,47 @@
+package com.example.bank.registration;
+
+import com.example.bank.exception.ClientNotFoundException;
+import com.example.bank.registration.jpa.Client;
+import com.example.bank.registration.jpa.ClientRepository;
+import jakarta.validation.constraints.Min;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequiredArgsConstructor
+/** 7) Dlaczego walidacja działa niezależnie od tego, czy w metodzie POST i PUT mam @Valid, czy też go tam nie ma??
+ * Podobno przed parametrem w controllerze powinien być @Valid. Usunąłem go i nadal działa.*/
+@Slf4j
+class ClientController {
+    private final ClientService clientService;
+    private final ClientRepository clientRepository;
```

Comment:
nie mozemy wstrzykiwac repozytoriow do kontrolerow. Psuje to flow calego kodu

Reply:
Już wiem. Poprawiłem.